OpenCV not found

I'm trying to compile the library, following the "How to use" section instructions, but when i run `make solib -j4`, I get that kind of error : 

> include/vilib/storage/subframe.h:25:32: fatal error: opencv2/core/mat.hpp: No such file or directory

I am using OpenCV 4.1.0 that is compiled from sources. As a consequence, I set these Makefile variables

```
CUSTOM_OPENCV_SUPPORT=1
CUSTOM_OPENCV_INSTALLATION_PATH=/usr/local/include/opencv4
```
Any idea why it is not working ? I think it might come from `CXX_NVCC_INCLUDES`, `CXX_LD_DIRS` or `CXX_LD_FLAGS` but I don't know where these should lead to.

--- a/README.md
+++ b/README.md
@@ -195,12 +195,12 @@ sudo apt-get install libjpeg-dev libpng-dev libtiff5-dev libdc1394-22-dev libeig
 # OpenCV
 git clone https://github.com/opencv/opencv.git
 cd opencv
-git checkout 4.2.0
+git checkout 4.3.0
 cd ..
 # OpenCV-Contrib
 git clone https://github.com/opencv/opencv_contrib.git
 cd opencv_contrib
-git checkout 4.2.0
+git checkout 4.3.0
 cd ..
 
 #
@@ -208,11 +208,8 @@ cd ..
 mkdir -p opencv/build
 cd opencv/build
 # Configure the build parameters
-#
-# Note to future self on CUDA_GENERATION:
-# https://github.com/opencv/opencv/blob/master/cmake/OpenCVDetectCUDA.cmake#L79
 cmake -D CMAKE_BUILD_TYPE=RELEASE \
-      -D CMAKE_INSTALL_PREFIX=/usr/local/opencv-4.2 \
+      -D CMAKE_INSTALL_PREFIX=/usr/local/opencv-4.3 \
       -D INSTALL_C_EXAMPLES=OFF \
       -D INSTALL_PYTHON_EXAMPLES=OFF \
       -D BUILD_EXAMPLES=OFF \
@@ -222,13 +219,8 @@ cmake -D CMAKE_BUILD_TYPE=RELEASE \
       -D WITH_V4L=ON \
       -D WITH_QT=OFF \
       -D WITH_OPENGL=ON \
-      -D WITH_OPENCL=ON \
-      -D WITH_CUDA=ON \
-      -D CUDA_GENERATION=Auto \
-      -D WITH_CUBLAS=ON \
-      -D WITH_CUFFT=ON \
-      -D WITH_NVCUVID=OFF \
-      -D CUDA_FAST_MATH=1 \
+      -D WITH_OPENCL=OFF \
+      -D WITH_CUDA=OFF \
       -D OPENCV_EXTRA_MODULES_PATH=../../opencv_contrib/modules ..
 # Start building
 # - use your number of CPU cores
@@ -236,7 +228,7 @@ make -j4
 # ..and "install" OpenCV
 sudo make install
 # create symlink (in order to support multiple installations)
-sudo ln -s /usr/local/opencv-4.2 /usr/local/opencv
+sudo ln -s /usr/local/opencv-4.3 /usr/local/opencv
 ```
 
 More information about the library is available [here](https://opencv.org/).

--- a/visual_lib/Makefile
+++ b/visual_lib/Makefile
@@ -134,12 +134,12 @@ DIRECTORIES=$(sort $(dir $(OBJECTS)))
 
 # Custom OpenCV support
 ifeq (${CUSTOM_OPENCV_SUPPORT},1)
-CXX_DEFINES+=-DCUSTOM_OPENCV_SUPPORT
-CXX_NVCC_INCLUDES+=-I${CUSTOM_OPENCV_INSTALLATION_PATH}/include
+CXX_NVCC_INCLUDES+=-I${CUSTOM_OPENCV_INSTALLATION_PATH}/include          \
+                   -I${CUSTOM_OPENCV_INSTALLATION_PATH}/include/opencv4
 CXX_LD_DIRS+=-L${CUSTOM_OPENCV_INSTALLATION_PATH}/lib
 CXX_LD_FLAGS+=-Wl,-rpath,${CUSTOM_OPENCV_INSTALLATION_PATH}/lib
-CXX_LD_LIBRARIES+=-lopencv_imgproc -lopencv_highgui -lopencv_core              \
-                  -lopencv_imgcodecs -lopencv_cudawarping
+CXX_LD_LIBRARIES+=-lopencv_imgproc -lopencv_highgui -lopencv_core  \
+                  -lopencv_imgcodecs -lopencv_features2d
 else
 CXX_LD_LIBRARIES+=$(shell pkg-config --libs-only-l opencv)
 endif

--- a/visual_lib/test/images/create_feature_detector_evaluation_data.sh
+++ b/visual_lib/test/images/create_feature_detector_evaluation_data.sh
@@ -47,8 +47,6 @@ import sys
 import os
 import cv2
 from cv_bridge import CvBridge
-from PIL import Image
-import numpy as np
 
 parser = argparse.ArgumentParser(description='Extract left and right images from captured bags. The script upon request concatenates the left and right images for easier checking.')
 parser.add_argument('bag', metavar='bag', help='ROS bag file to be analyzed')

--- a/visual_lib/test/include/test/test_base.h
+++ b/visual_lib/test/include/test/test_base.h
@@ -74,7 +74,8 @@ protected:
                                const std::vector<cv::Mat> & image_pyramid2,
                                unsigned int diff_threshold) const;
     bool is_list_file(const char * file_path);
-    void load_image_dimensions(const std::size_t & width_default = 0,
+    bool verify_path(const char * path);
+    bool load_image_dimensions(const std::size_t & width_default = 0,
                                const std::size_t & height_default = 0);
 
     const char * name_;

--- a/visual_lib/test/src/feature_detection/test_fast.cpp
+++ b/visual_lib/test/src/feature_detection/test_fast.cpp
@@ -86,7 +86,10 @@ bool TestFAST::run(void) {
   stat_gpu.emplace_back("[1]","FAST feature count");
 
   // Create detectors
-  load_image_dimensions();
+  if(!load_image_dimensions()) {
+    // Could not acquire the initialization parameters
+    return false;
+  }
   // CPU
 #if ENABLE_CPU_VERSION
   detector_cpu_.reset(new rosten::FASTCPU<false>(

--- a/visual_lib/test/src/high_level/test_featuretracker.cpp
+++ b/visual_lib/test/src/high_level/test_featuretracker.cpp
@@ -78,7 +78,10 @@ bool TestFeatureTracker::run(void) {
     feature_tracker_options.affine_est_offset = false;
 
     // Instantiate detectors (default is the Blender dataset)
-    load_image_dimensions(752,480);
+    if(!load_image_dimensions(752,480)) {
+      // Could not acquire the initialization parameters
+      return false;
+    }
     // Create feature detector & tracker for the GPU
     detector_gpu_.reset(new FASTGPU(image_width_,
                                     image_height_,

--- a/visual_lib/test/src/high_level/test_featuretracker.cpp
+++ b/visual_lib/test/src/high_level/test_featuretracker.cpp
@@ -289,7 +289,7 @@ void TestFeatureTracker::calculate_feature_error(
                             Eigen::Vector4d,
                             std::hash<std::size_t>,
                             std::equal_to<std::size_t>,
-                            Eigen::aligned_allocator<Eigen::Vector4d>> track_origin_point;
+                            Eigen::aligned_allocator<std::pair<const std::size_t,Eigen::Vector4d>>> track_origin_point;
   double tracked_points_rmse = 0.0;
   std::size_t tracked_points = 0;
   for(std::size_t i=0;i<frame->num_features_;++i) {

--- a/visual_lib/test/src/test_base.cpp
+++ b/visual_lib/test/src/test_base.cpp
@@ -21,6 +21,8 @@
 
 #include <iostream>
 #include <fstream>
+#include <sys/types.h>
+#include <sys/stat.h>
 #include <opencv2/highgui.hpp>
 #include "test/test_base.h"
 #include "test/common.h"
@@ -67,8 +69,12 @@ void TestBase::load_image(int load_flags, bool display_image, bool display_info)
   }
 }
 
-void TestBase::load_image_dimensions(const std::size_t & width_default,
+bool TestBase::load_image_dimensions(const std::size_t & width_default,
                                      const std::size_t & height_default) {
+  if(!verify_path(file_path_)) {
+    std::cout << " The specified file does not exist (" << file_path_ << ")" << std::endl;
+    return false;
+  }
   if(is_list_) {
     // this is a list file, read out the resolution from the first 2 lines
     std::ifstream list_file(file_path_);
@@ -98,6 +104,7 @@ void TestBase::load_image_dimensions(const std::size_t & width_default,
     image_width_ = width_default;
     image_height_ = height_default;
   }
+  return true;
 }
 
 void TestBase::load_image_to(const char * image_path,
@@ -221,6 +228,11 @@ bool TestBase::is_list_file(const char * file_path) {
     }
   }
   return false;
+}
+
+bool TestBase::verify_path(const char * path) {
+  struct stat info;
+  return (stat(path,&info) == 0);
 }
 
 bool TestBase::run_benchmark_suite(std::vector<Statistics> & stat_cpu,

--- a/visual_lib/test/src/tests.cpp
+++ b/visual_lib/test/src/tests.cpp
@@ -22,9 +22,6 @@
 #include <iostream>
 #include <vector>
 #include <memory>
-#ifdef CUSTOM_OPENCV_SUPPORT
-#include <opencv2/cudev/common.hpp>
-#endif /* CUSTOM_OPENCV_SUPPORT */
 #include "vilib/cuda_common.h"
 #include "test/common.h"
 #include "test/test_base.h"
@@ -59,19 +56,6 @@ using namespace vilib;
 int main(int argc, char * argv[]) {
   // Save arguments
   init_arguments(argc,argv);
-
-  // Initializations
-#ifdef CUSTOM_OPENCV_SUPPORT
-  /*
-   * Placing cv::Mats in page-locked memory buffers
-   * The CUDA memcpy-s became super-fast, but operations on the matrices on the CPU got somewhat slower:
-   * - the copyies got fast: because we dont need to copy the area to a
-   *   pinned location first, hence we save the first step of a regular H2D,D2H transfer
-   * - operations on the CPU got slower, because the pinned-memory content is NOT
-   *   cached on the CPU
-   */
-  cv::Mat::setDefaultAllocator(cv::cuda::HostMem::getAllocator(cv::cuda::HostMem::AllocType::PAGE_LOCKED));
-#endif /* CUSTOM_OPENCV_SUPPORT */
 
   /*
    * Note to future self:

